doc/api/crypto.md

```diff
@@ -4165,6 +4165,16 @@ const {
 console.log(getHashes()); // ['DSA', 'DSA-SHA', 'DSA-SHA1', ...]
 ```
 
+### `crypto.getOpenSSLSecLevel()`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* Returns: {number} The [`default OpenSSL security level`][].
+  Useful to be able to adjust based on the algorithms available given the
+  security level.
```

Comment:
Some explanation as to how to interpret the returned values would be helpful. 

And just for my knowledge because I'm not that familiar with the security levels... are the return values specific at all the version of OpenSSL? Is it supported by BoringSSL?

Reply:
Ok just digging into the API a bit more ... https://docs.openssl.org/3.0/man3/SSL_CTX_set_security_level/#description 

Looks like there are six security levels 0 through 5.... honestly we should document these and/or link to the relevant openssl definitions of them here.

Reply:
> And just for my knowledge because I'm not that familiar with the security levels... are the return values specific at all the version of OpenSSL? Is it supported by BoringSSL?

Yes, the values mean different things for different OpenSSL versions.

Compare the level descriptions for 1.1.1, 3.0 and 3.4:
- https://docs.openssl.org/1.1.1/man3/SSL_CTX_set_security_level/#default-callback-behaviour
- https://docs.openssl.org/3.0/man3/SSL_CTX_set_security_level/#default-callback-behaviour
- https://docs.openssl.org/3.4/man3/SSL_CTX_set_security_level/#default-callback-behaviour

Reply:
This makes me more nervous about this API as users will need to know exactly which version of openssl the version of node.js they are using happens to have been compiled with. This absolutely needs to be documented and explained.

Reply:
Maybe this should be an internal only API that is used for the tests?

There's definite value in being to get this information and use it to adjust test expectations to make it easier for downstream rebuilders of Node.js (e.g. Linux distros) to run the tests when Node.js is linked to a system OpenSSL. I'm less convinced of the usefulness for general Node.js applications.

Reply:
I'd be happy with that. I really can't think of much of a general use case for this outside of our tests.